test/runtests.jl

```diff
@@ -283,6 +283,13 @@ end
         @test !isnothing(i)
         @test is_no_escape(escapes.ssavalues[i])
     end
+
+    let
+        src, escapes = analyze_escapes((Ref{String},)) do a
+            return @isdefined(b)
```

Comment:
Hm, I meant in this case the compiler should have elided this `:isdefined` check already at this point..
I'm wondering what are the cases we really need to handle.

Reply:
let me record the exprs you presented in #23 and see what's the case of `isdefined`.

Reply:
Yeah, that would be a good starting point of investigation.

Reply:
I did the experiment and also verify the testcase, finding that actually in this case the `:isdefined` expr is not elided:
```
julia> using EscapeAnalysis

julia> function f(a::String)
           return @isdefined(b)
       end
f (generic function with 1 method)

julia> src, escapes = @analyze_escapes f("haha")
typeof(f)(_2::String ↑)
1 ◌ 1 ─ %1 = $(Expr(:isdefined, :(Main.b)))::Bool
2 ◌ └──      return %1
```

Reply:
If `:isdefined` only deals with global objects, then it's not our interest ? When `:isdefined` works on objects within local scope ?

Reply:
There are two types of `isdefined` in my collected results of #23 :
```
- [ ] `:$(Expr(:isdefined, :(%5)))`
- [ ] `:$(Expr(:isdefined, :($(Expr(:static_parameter, 1)))))`
```
the first one is clearly an `isdefined` on local scope, the second I am not sure about it.

Reply:
We may want to have either of those cases as a test case here, but it's also obvious that `:isdefined` never escapes anything, so I'm okay to merge this as is if you want.

Reply:
I will add another case to cover the local scope case.

Reply:
added